sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java

```diff
@@ -216,53 +216,37 @@ public AccessToken authenticateWithConfidentialClientCache(TokenRequestContext r
     @SuppressWarnings("deprecation")
     public MsalToken authenticateWithPublicClientCache(TokenRequestContext request, IAccount account) {
         PublicClientApplication pc =  getPublicClientInstance(request).getValue();
+        MsalToken token = acquireTokenFromPublicClientSilently(request, pc, account, false);
+        if (OffsetDateTime.now().isAfter(token.getExpiresAt().minus(REFRESH_OFFSET))) {
+            token = acquireTokenFromPublicClientSilently(request, pc, account, true);
+        }
+        return token;
+    }
+
+    private MsalToken acquireTokenFromPublicClientSilently(TokenRequestContext request,
+                                                           PublicClientApplication pc,
+                                                           IAccount account,
+                                                           boolean forceRefresh) {
         SilentParameters.SilentParametersBuilder parametersBuilder = SilentParameters.builder(
             new HashSet<>(request.getScopes()));
 
-        if (request.getClaims() != null) {
-            ClaimsRequest customClaimRequest = CustomClaimRequest.formatAsClaimsRequest(request.getClaims());
-            parametersBuilder.claims(customClaimRequest);
+        if (forceRefresh) {
             parametersBuilder.forceRefresh(true);
         }
 
-        if (account != null) {
-            parametersBuilder = parametersBuilder.account(account);
-        }
-        parametersBuilder.tenant(
-            IdentityUtil.resolveTenantId(tenantId, request, options));
-        try {
-            MsalToken accessToken = new MsalToken(pc.acquireTokenSilently(parametersBuilder.build()).get());
-            if (OffsetDateTime.now().isBefore(accessToken.getExpiresAt().minus(REFRESH_OFFSET))) {
-                return accessToken;
-            }
-        } catch (MalformedURLException e) {
-            throw LOGGER.logExceptionAsError(new RuntimeException(e.getMessage(), e));
-        } catch (ExecutionException | InterruptedException e) {
-            // Cache misses should not throw an exception, but should log.
-            if (e.getMessage().contains("Token not found in the cache")) {
-                LOGGER.verbose("Token not found in the MSAL cache.");
-                return null;
-            } else {
-                throw LOGGER.logExceptionAsError(new ClientAuthenticationException(e.getMessage(), null, e));
-            }
-        }
-
-        SilentParameters.SilentParametersBuilder forceParametersBuilder = SilentParameters.builder(
-            new HashSet<>(request.getScopes())).forceRefresh(true);
-
         if (request.isCaeEnabled() && request.getClaims() != null) {
             ClaimsRequest customClaimRequest = CustomClaimRequest
                 .formatAsClaimsRequest(request.getClaims());
-            forceParametersBuilder.claims(customClaimRequest);
+            parametersBuilder.claims(customClaimRequest);
```

Comment:
forceRefresh should set to true here ?

Reply:
It is set true in the conditional on line 233.

Reply:
The equivalent async flow is setting it to true here:

https://github.com/Azure/azure-sdk-for-java/pull/38792/files/dfb75b8dad25e94487f25c5e93972a906085900f#diff-0983e9433b1bbd62e50b6b0546ed7f9d694351e56249a8af0683b9eafa208713R638